src/Build/BackEnd/Shared/BuildRequestConfiguration.cs

```diff
@@ -299,7 +299,11 @@ public bool IsTraversal
             {
                 if (!_isTraversalProject.HasValue)
                 {
-                    if (String.Equals(Path.GetFileName(ProjectFullPath), "dirs.proj", StringComparison.OrdinalIgnoreCase))
+#if NET471_OR_GREATER
+                    if (MemoryExtensions.Equals(Microsoft.IO.Path.GetFileName(ProjectFullPath.AsSpan()), "dirs.proj".AsSpan(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Are you doing this explicitly here to be conservative/localized (totally legit!) or because there's a reason not to shift to Microsoft.IO.Redist elsewhere in this file?

Reply:
There can be subtle differences in behavior between NetFX System.IO.Path and Microsoft.IO.Path, so this was meant to be a surgical update to minimize churn. In other repositories, we've seen measurable improvement fully switching over to Microsoft.IO, so this should be considered as well.

One thing to note: Since Microsoft.IO is a snapshot of improvements backported for NetFX, we want to use System.IO in .NET 5+. That way we'll continue to benefit from future improvements.